certify-service/src/main/java/io/mosip/certify/services/CertifyIssuanceServiceImpl.java

```diff
@@ -157,13 +160,22 @@ private VCResult<?> getVerifiableCredential(CredentialRequest credentialRequest,
                     String templatedVC = vcFormatter.format(identityData, templateParams);
                     Map<String, String> vcSignerParams = new HashMap<>();
                     // TODO: Collate this into simpler APIs where just key-type is specified
-                    vcSignerParams.put(KeyManagerConstants.VC_SIGN_ALGO,
-                            SignatureAlg.RSA_SIGNATURE_SUITE);
-                    vcSignerParams.put(KeyManagerConstants.PUBLIC_KEY_URL, hostedKey);
-                    vcSignerParams.put(KeyManagerConstants.KEY_APP_ID, Constants.CERTIFY_MOCK_RSA);
-                    vcSignerParams.put(KeyManagerConstants.KEY_REF_ID, Constants.EMPTY_REF_ID);
-                    // Change it to PS256 as per --> https://w3c.github.io/vc-jws-2020/#dfn-jsonwebsignature2020
-                    vcSignerParams.put(KeyManagerConstants.KEYMGR_SIGN_ALGO, JWSAlgorithm.RS256.getName());
+                    if (VCSignAlgo.equals(SignatureAlg.RSA_SIGNATURE_SUITE)) {
```

Comment:
can we move this from if else to a different pattern

Reply:
Added a ticket for the above --> https://mosip.atlassian.net/browse/INJICERT-570